test/test_qdmi.cpp

```diff
@@ -970,3 +971,100 @@ TEST_P(QDMIImplementationTest, QueryPulseSupportLevel) {
   EXPECT_EQ(ret, QDMI_SUCCESS);
   EXPECT_EQ(pulse_support_level, QDMI_DEVICE_PULSE_SUPPORT_LEVEL_NONE);
 }
+
+TEST_P(QDMIImplementationTest, QueryEveryTelemetryProperties) {
+
+  const auto fomac = FoMaC(device);
+
+  auto telemetries = fomac.get_telemetry_sensors();
+
+  ASSERT_GT(telemetries.size(), 0);
+
+  for (const auto &telemetry : telemetries) {
+
+    auto telemetry_id = fomac.get_telemetry_id(telemetry);
+    EXPECT_STRNE(telemetry_id.c_str(), "");
+
+    auto telemetry_unit = fomac.get_telemetry_unit(telemetry);
+    EXPECT_STRNE(telemetry_unit.c_str(), "");
+
+    auto sampling_rate = fomac.get_telemetry_sampling_rate(telemetry);
+    EXPECT_GT(sampling_rate, 0);
+  }
+}
+
+TEST_P(QDMIImplementationTest, TelemetryQuery) {
+  if (mode == TEST_SESSION_MODE::READONLY) {
+    GTEST_SKIP() << "Skipping test for read-only session";
+  }
+
+  const auto fomac = FoMaC(device);
+
+  const std::vector<QDMI_TelemetrySensor> telemetry_sensors =
+      fomac.get_telemetry_sensors();
+
+  ASSERT_GT(telemetry_sensors.size(), 0);
+
+  for (QDMI_TelemetrySensor telemetry_sensor : telemetry_sensors) {
+    QDMI_TelemetrySensor_Query query = nullptr;
+    QDMI_TelemetrySensor_Query_Status status = {};
+    time_t start_time = time(&start_time);
+    time_t end_time = time(&end_time) + 600;
```

Comment:
This is uncommon usage of the `time` function. I would recommend:
```suggestion
    time_t start_time = time(nullptr);
    time_t end_time = time(nullptr) + 600;
```